addons/mail/models/mail_message.py

```diff
@@ -679,6 +680,8 @@ def _update_content(self, body, attachment_ids):
         thread = self.env[self.model].browse(self.res_id)
         thread._check_can_update_message_content(self)
         self.body = body
+        if self._is_empty_message():
+            self._clean_related_data()
```

Comment:
should be done after deleting the attachments as this is part of the condition for emptyness